Pcap++/src/PcapFileDevice.cpp

```diff
@@ -55,6 +55,18 @@ std::string IFileDevice::getFileName() const
 	return m_FileName;
 }
 
+bool IFileDevice::isTimestampPrecisionNanoSupported()
+{
+#if defined(PCAP_TSTAMP_PRECISION_NANO)
+	return true;
+#else
+	PCPP_LOG_DEBUG(
+		"PcapPlusPlus was compiled without nano precision support which requires libpcap > 1.5.1. Please "
+		"recompile PcapPlusPlus with nano precision support to use this feature. Using default microsecond precision");
+	return false;
+#endif
+}
```

Comment:
If there are no compilation errors, it can be great. I'll check it but not sure because if i'm not mistaken some functions introduced after 1.5.1. So, we will probably see some "undefined ..." errors

Reply:
https://github.com/seladb/PcapPlusPlus/blob/6d17ca81fbc80dfd4fa3529b18ed81a753f192f3/Pcap%2B%2B/src/PcapFileDevice.cpp#L255 

This function is introduced with nanosecond precision support in `libpcap`. So, unfortunately all checks must be preprocessor directives